gameplay.py

```diff
@@ -2,7 +2,17 @@
 import random
 from typing import Optional
 
-from models import Pokemon, Move, DamageClass, Type, MoveInfo, PokemonSpecies, Trainer, PokemonStatus
+from models import (
+    Pokemon, Move, DamageClass, Type, MoveInfo, PokemonSpecies, Trainer, PokemonStatus, Ailment
+)
+
+NON_VOLATILE_AILMENTS = {Ailment.BURN, Ailment.PARALYSIS, Ailment.POISON}
```

Comment:
I would make this a property on the Ailment and PokemonStatus enums: ie `Ailment.BURN.volatile`:

it'd look something like this:

```python
@property
def volatile(self):
    return self in {
        Ailment.BURN,
        Ailment.PARALYSIS,
        Ailment.POISON
    }
```

and 

```python
@property
def volatile(self):
     return self in {
       PokemonStatus.BURNED,
       PokemonStatus.PARALYZED,
       PokemonStatus.POISONED,
       PokemonStatus.BADLY_POISONED  
     }
```

and then make use of it in the apply_ailment function

ie

```python
defending_pokemon.statuses & NON_VOLATILE_STATUSES and ailment in NON_VOLATILE_AILMENTS
# To:
any(not status.volatile for status in defending_pokemon.statuses) and not ailment.volatile
```